src/averageAges.js

```diff
@@ -1,61 +1,57 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let mens = [];
+
+  century ? (
+    mens = people.filter(person => person.sex === 'm'
+    && Math.ceil(person.died / 100) === century)
+  ) : mens = people.filter(person => person.sex === 'm');
+
+  const mensAvgAge = mens
+    .map(men => men.died - men.born)
+    .reduce((totalYears, menAge) => totalYears + menAge, 0);
+
+  return Math.round(mensAvgAge / mens.length * 100) / 100 || 0;
```

Comment:
you do the same on calculateWomenAverageAge. You can create separate function and you it here and on calculateWomenAverageAge